core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -476,6 +477,14 @@ class CivilizationInfo {
         if (tech.freeTechs != 0) return true
         return tech.currentTechnology() == null && cities.isNotEmpty()
     }
+    
+    fun getEquivalentConstructionName(constructionName: String): String? {
```

Comment:
I don't see a situation where this is genericized, you can just use getEquivalentUnit directly in the place you changed. From what I see this just changes a direct call to a more ambiguous call.